### ISSUE TITLE
Configurable certificate directory

### ISSUE THREAD

**Issue by isuldor** [decred/dcrstakepool#585]
dcrstakepool doesn't have a data directory (#502), and as a side effect will always create and use a `.dcrstakepool` directory under the user's home directory. This is where the daemon expects stakepoold certificates to exist.  Several path config variables already exist for other resources: `--publicpath`, `--templatepath`, and `--configfile`.  Perhaps a `--certpath` option is in order?

An alternative to setting all of these separate path variables (when not using `~/.dcrstakepool`) would be a home directory option that is comparable to the deprecated `--appdata`.  Maybe `--homedir`?

**Comment by JoeGruffins** [comment]
Such a setting would be useful for an operator? I can see where it would help if you were running a testnet site from the same machine, is that where you are coming from?

**Comment by isuldor** [comment]
Yeah the usefulness we're talking about here is probably really mild.  But I think it's wise to run dcrstakepool as a user with restricted permissions (and a false shell like `/usr/bin/nologin`). Personally I like to pick a home directory under under `/opt`, although some might prefer `/var` or something else.  That results in the certificates needing to be in `/opt/dcrstakepool/.dcrstakepool`. An ugly workaround would be to specify the full cert path in the config file like: 

`stakepooldcerts=/opt/dcrstakepool/certs/node01.cert,/opt/dcrstakepool/certs/node02.cert,/opt/dcrstakepool/certs/node03.cert,/opt/dcrstakepool/certs/node04.cert,/opt/dcrstakepool/certs/node05.cert,/opt/dcrstakepool/certs/node06.cert`

Another consideration is if this software is ever distributed via a package manager, where it would be useful to configure any of the necessary paths.  A single --homedir fits nicely in that case.  For reference, I think [politeia](https://github.com/decred/politeia/search?q=HomeDir&type=Code) uses a homedir variable.

**Comment by JoeGruffins** [comment]
Sounds like good reasons to me.  @jholdstock any thoughts?
I need to look how the other repo's do it, but I guess it make sense to be a command line only command? If the conf is also located in that directory.
